contracts/TokenFactory.sol

```diff
@@ -136,30 +145,27 @@ contract TokenFactory is ReentrancyGuard, Ownable {
         token.initialize(name, symbol, uri, address(this));
         tokens[tokenAddress] = TokenState.FUNDING;
 
-        tokensAddresses[totalTokensAddresses] = tokenAddress;
-        totalTokensAddresses++;
+        uint _totalTokensByCompetitionId = totalTokensByCompetitionId[currentCompetitionId];
+
+        tokensByCompetitionId[currentCompetitionId][_totalTokensByCompetitionId] = tokenAddress;
```

Comment:
can define the mapping as mapping(uint256 => address[]) instead, and use `.push` to add new address, and .length to get total tokens for the competition id